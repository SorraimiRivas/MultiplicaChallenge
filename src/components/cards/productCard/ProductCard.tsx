import {View, Text, Image, Pressable} from 'react-native';
import React, {FC} from 'react';
import styles from './style';

type Props = {
  productName: string;
  createdAt: string;
  points: number;
  image: string;
  isRedemption: boolean;
  onPress: () => void;
};

const ProductCard: FC<Props> = ({
  onPress,
  productName,
  points,
  createdAt,
  image,
  isRedemption,
}) => {
  const signColor = isRedemption ? 'red' : 'green';

  return (
    <Pressable style={styles.cardContainer} onPress={onPress}>
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={{
            uri: image,
          }}
        />
      </View>
      <View>
        <Text style={styles.productName}>{productName}</Text>
        <Text style={styles.createdAt}>{createdAt}</Text>
      </View>
      <View style={styles.pointsContainer}>
        <Text style={{color: signColor, fontWeight: '800'}}>
          {isRedemption ? '- ' : '+ '}
          <Text style={styles.points}>{points}</Text>
        </Text>
      </View>
      <View>
        <Text style={styles.arrow}>{'\u003e'}</Text>
      </View>
    </Pressable>
  );
};

export default ProductCard;
