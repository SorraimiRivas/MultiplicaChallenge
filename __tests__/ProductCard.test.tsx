import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from '../src/components/cards/productCard/ProductCard';

test('renders ProductCard component', () => {
  const card = renderer
    .create(
      <ProductCard
        createdAt="3 de Marzo, 1995"
        points={100}
        isRedeemed={false}
        onPress={() => null}
        productName="Name"
        image="https://i.imgur.com/3l7u1Z2.jpg"
      />,
    )
    .toJSON();

  expect(card).toMatchSnapshot();
});
