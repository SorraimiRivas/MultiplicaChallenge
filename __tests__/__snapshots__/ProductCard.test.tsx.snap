// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductCard component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": 55,
      "opacity": 1,
      "width": 333,
    }
  }
>
  <View
    style={
      {
        "height": 55,
        "width": 55,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://i.imgur.com/3l7u1Z2.jpg",
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 45,
          "resizeMode": "cover",
          "width": 45,
        }
      }
    />
  </View>
  <View
    style={
      {
        "bottom": 5,
        "marginRight": 30,
        "width": 175,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 14,
          "fontWeight": "800",
        }
      }
    >
      Name
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 12,
          "fontWeight": "400",
        }
      }
    >
      -/-/-
    </Text>
  </View>
  <View
    style={
      {
        "bottom": 5,
        "right": 25,
        "width": 60,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Avenir",
            "fontSize": 16,
            "fontWeight": "900",
          },
          {
            "color": "green",
          },
        ]
      }
    >
      +
      <Text
        style={
          {
            "color": "#000000",
            "fontFamily": "Avenir",
            "fontSize": 16,
            "fontWeight": "800",
          }
        }
      >
        100
      </Text>
    </Text>
  </View>
  <View>
    <Text
      style={
        {
          "bottom": 5,
          "color": "#000000",
          "fontFamily": "Avenir",
          "fontSize": 20,
          "fontWeight": "800",
          "right": 5,
        }
      }
    >
      &gt;
    </Text>
  </View>
</View>
`;
